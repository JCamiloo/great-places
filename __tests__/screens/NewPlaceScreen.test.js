import "react-native";
import React from "react";
import NewPlaceScreen from "../../screens/NewPlaceScreen";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureMockStore();
const store = mockStore({});

describe("NewPlaceScreen tests", () => {
  test("NewPlaceScreen renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <NewPlaceScreen navigation={{ getParam: jest.fn() }} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
