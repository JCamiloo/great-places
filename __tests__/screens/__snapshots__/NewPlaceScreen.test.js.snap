// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPlaceScreen component tests NewPlaceScreen component renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "margin": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "marginBottom": 15,
          }
        }
      >
        Title
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "marginBottom": 15,
            "paddingHorizontal": 2,
            "paddingVertical": 4,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 200,
              "justifyContent": "center",
              "marginBottom": 10,
              "width": "100%",
            }
          }
        >
          <Text>
            No image picked yet.
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#fc9208",
                  },
                ]
              }
            >
              Take image
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 150,
              "justifyContent": "center",
              "marginBottom": 10,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text>
            No location chosen yet
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#fc9208",
                    },
                  ]
                }
              >
                Get user location
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#fc9208",
                    },
                  ]
                }
              >
                Pick on Map
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#fc9208",
                },
              ]
            }
          >
            Save Place
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`NewPlaceScreen tests NewPlaceScreen renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "margin": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "marginBottom": 15,
          }
        }
      >
        Title
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "marginBottom": 15,
            "paddingHorizontal": 2,
            "paddingVertical": 4,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 200,
              "justifyContent": "center",
              "marginBottom": 10,
              "width": "100%",
            }
          }
        >
          <Text>
            No image picked yet.
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#fc9208",
                  },
                ]
              }
            >
              Take image
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#ccc",
              "borderWidth": 1,
              "height": 150,
              "justifyContent": "center",
              "marginBottom": 10,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text>
            No location chosen yet
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#fc9208",
                    },
                  ]
                }
              >
                Get user location
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#fc9208",
                    },
                  ]
                }
              >
                Pick on Map
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#fc9208",
                },
              ]
            }
          >
            Save Place
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
